src/interpreter/executors/instruction.rs

```diff
@@ -111,11 +111,17 @@ where
             OpcodeRepr::EXP => {
                 self.gas_charge(GAS_EXP)?;
                 self.alu_boolean_overflow(ra, Word::overflowing_pow, b, c as u32)?;
+                if self.registers[REG_OF] != 0 {
```

Comment:
Shouldn't this be a behavior of `alu_*_overflow` instead? Also, we don't put concrete implementations here. If exp is a special case of `allu`, then we need to create a new `alu_*` function and call that instead

Reply:
Fixed